tests/functional/test_states.py

```diff
@@ -0,0 +1,93 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# todo add new DSL variant for job and multijob
+# todo add multistate tests for job and multijob
+
+# pylint:disable=redefined-outer-name, no-self-use
+import allure
+from adcm_client.objects import ADCMClient
+from adcm_pytest_plugin.utils import fixture_parametrized_by_data_subdirs
+from tests.functional.plugin_utils import build_objects_checker
+
+
+ACTION_NAME = 'state_changing_action'
+
+
+@fixture_parametrized_by_data_subdirs(__file__, 'cluster_and_service')
+def cluster(sdk_client_fs: ADCMClient, request):
+    bundle = sdk_client_fs.upload_from_fs(request.param)
+    cluster = bundle.cluster_create(name=bundle.name)
+    cluster.service_add(name='first_srv')
+    check_objects_state_changed = build_objects_checker(
+        field_name='State',
+        changed=cluster.name,
+        extractor=lambda obj: obj.state,
+    )
+    return cluster, check_objects_state_changed
```

Comment:
As this method returns not only a cluster, it should have type hints and maybe docstring to clarify why it does so

Reply:
And unpacking of this value looks confusing. Rename it?

Reply:
Hints were added. Need ideas about naming

Reply:
I have just basic ideas: `cluster_and_checker`, `cluster_and_objects_checker`.
Or maybe just split into two fixtures, especially as the `cluster` value is unpacked in each test.

Reply:
fixed